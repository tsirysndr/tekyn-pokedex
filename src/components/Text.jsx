import React from "react";
import PropTypes from "prop-types";
import { TouchableHighlight } from "react-native";
import styled from "@emotion/native";
import { useFonts, Roboto_500Medium } from "@expo-google-fonts/roboto";

const StyledText = styled.Text`
  color: ${props => props.color};
  font-family: 'Roboto_500Medium';
`;

export default function Text({ color, onPress, children }) {
  const [fontsLoaded] = useFonts({ Roboto_500Medium });
  if (!fontsLoaded) {
    return null;
  }
  return <StyledText color={color}>{children}</StyledText>;
}

Text.defaultProps = {
  children: null,
  color: "#fff",
};

Text.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
};
