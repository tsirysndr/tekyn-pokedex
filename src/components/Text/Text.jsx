import React from "react";
import PropTypes from "prop-types";
import { TouchableHighlight } from "react-native";
import styled, { css } from "@emotion/native";
import { useFonts, Roboto_500Medium } from "@expo-google-fonts/roboto";

const StyledText = styled.Text`
  color: ${(props) => props.color};
  ${({ fontsLoaded }) =>
    fontsLoaded &&
    css`
      font-family: "Roboto_500Medium";
    `}
`;

export default function Text(props) {
  const [fontsLoaded] = useFonts({ Roboto_500Medium });
  return (
    <StyledText {...props} fontsLoaded={fontsLoaded}>
      {props.children}
    </StyledText>
  );
}

Text.defaultProps = {
  children: null,
  color: "#fff",
};

Text.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
};
