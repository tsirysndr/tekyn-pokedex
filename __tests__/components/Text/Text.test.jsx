import Text from "@Components/Text";
import { render } from "@testing-library/react-native";

test("renders Text", () => {
  const tree = render(
    <Text>Hello</Text>
  );
  expect(tree).toMatchSnapshot();
});
