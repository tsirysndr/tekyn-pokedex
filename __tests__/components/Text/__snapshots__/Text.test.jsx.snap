// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Text 1`] = `
<Text
  color="#fff"
  fontsLoaded={false}
  style={
    Array [
      Object {
        "color": "#fff",
      },
      undefined,
    ]
  }
>
  Hello
</Text>
`;
